DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/Repositories/SchoolRepository.cs

```diff
@@ -19,4 +20,30 @@ public class SchoolRepository(IAcademiesDbContext academiesDbContext) : ISchoolR
                     : SchoolCategory.LaMaintainedSchool))
             .SingleOrDefaultAsync();
     }
+
+    public async Task<SchoolDetails?> GetSchoolDetailsAsync(int urn)
+    {
+       return await academiesDbContext.GiasEstablishments
+            .Where(e => e.Urn == urn)
+            .Select(establishment => new SchoolDetails(establishment.EstablishmentName!,
+                stringFormattingUtilities.BuildAddressString(
+                    establishment.Street,
+                    null,
+                    establishment.Town,
+                    establishment.Postcode
+                ),
+                establishment.GorName!,
+                establishment.LaName!,
+                establishment.PhaseOfEducationName!,
+                new AgeRange(establishment.StatutoryLowAge!, establishment.StatutoryHighAge!),
+                establishment.NurseryProvisionName))
+            .SingleOrDefaultAsync();
+    }
+
+    public async Task<DateOnly> GetDateJoinedTrustAsync(int urn)
+    {
+        return await academiesDbContext.GiasGroupLinks.Where(gl => gl.Urn == urn.ToString())
+            .Select(gl => DateOnly.ParseExact(gl.JoinedDate!, "dd/MM/yyyy", CultureInfo.InvariantCulture, DateTimeStyles.None))
+            .FirstAsync();
```

Comment:
```suggestion
            .SingleAsync();
```
There should only ever be one group link returned for a urn, if there are more then something fundamental will have changed in the db and we will have to think carefully about which one we return.